### PR TITLE
Implement the Prism DEM datasets, add tests.

### PR DESCRIPTION
**Pull request by BSchilperoort**
A demo notebook is included in demo/prism_dem_demo.ipynb

### CODE REVIEW

**Review by geek-yang — APPROVED** [review]
Nice work @BSchilperoort . Very clean implementation. One small issue about the recipe. If you fix the time thingy for xarray (just like canopy-height dataset), then everything should work. 

Tbh, at the beginning I thought the implementation of prism is the same as canopy and many things can be merged. But then when diving into the details, just notice that many things are actually different. Devils in the details 😂.

**Comment by geek-yang** [comment]
The API works nicely. But when I tried with the recipe, an error occur:

```sh
~/EcoExtreML/temp via 🐍 v3.10.12 (ecoextreml) took 2s
❯ zampy prism90_dem_recipe.yml
Executing recipe: prism90_dem_recipe.yml
Copernicus_DSM_30_N50_00_E003_00.tar: 8.97MB [00:01, 5.29MB/s]
elevation renamed to elevation.
Conversion of dataset 'prism-dem-90' following ALMA convention is complete!
Traceback (most recent call last):
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/xarray/core/dataset.py", line 1382, in _construct_dataarray
    variable = self._variables[name]
KeyError: 'time'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/yangliu/venv/ecoextreml/bin/zampy", line 8, in <module>
    sys.exit(run_recipe())
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/yangliu/EcoExtreML/zampy/src/zampy/cli.py", line 13, in run_recipe
    rm.run()
  File "/home/yangliu/EcoExtreML/zampy/src/zampy/recipe.py", line 121, in run
    ds = ds.resample(time=self.frequency).mean()
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/xarray/core/dataset.py", line 9322, in resample
    return self._resample(
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/xarray/core/common.py", line 982, in _resample
    dim_coord = self[dim]
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/xarray/core/dataset.py", line 1473, in __getitem__
    return self._construct_dataarray(key)
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/xarray/core/dataset.py", line 1384, in _construct_dataarray
    _, name, variable = _get_virtual_variable(self._variables, name, self.dims)
  File "/home/yangliu/venv/ecoextreml/lib/python3.10/site-packages/xarray/core/dataset.py", line 196, in _get_virtual_variable
    raise KeyError(key)
KeyError: 'time'
```

Clearly it is due to the missing time axis, which is required in the recipe for saving the output after conversion. But it is easy to fix, just add a time coord, similar to canopy dataset.

**Comment by BSchilperoort** [comment]
@geek-yang thanks for the review! I had fixed the missing time dim issue, but seem to have forgotten to commit it :man_facepalming:. Instead of adding a time coordinate, I modify `recipe.py` to allow for variables without a time dim.

File: src/zampy/datasets/prism_dem.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    # The prism DEM variable & coords already follow the CF/Zampy convention
+    for variable in ds.variables:
+        variable_name = str(variable)  # Cast to string to please mypy.
+        if variable_name != "time":
+            ds[variable_name].attrs["units"] = str(
+                VARIABLE_REFERENCE_LOOKUP[variable_name].unit
+            )
+            ds[variable_name].attrs["description"] = VARIABLE_REFERENCE_LOOKUP[
+                variable_name
+            ].desc
```
[the remark this answers is not in the document]
Comment (BSchilperoort) [reply src/zampy/datasets/prism_dem.py:261]: Ah yeah, this does overlap. We can probably make it more generic and wrap it into a separate (util) function, however that might be best left for when we make the whole attribute writing more complete.

[the remark this answers is not in the document]
Comment (geek-yang) [reply src/zampy/datasets/prism_dem.py:261]: That's also my thought. Great!

**Pull request merged**